### PR TITLE
bug-fix: Fixed a bug where channel is being removed from my channel list when other member leaves the channel

### PR DESCRIPTION
**Pull request by liamcho**
Fixes: [CLNP-2121](https://sendbird.atlassian.net/browse/CLNP-2121)

### Original PR
- https://github.com/sendbird/sendbird-uikit-react/pull/937

### Changelogs
- Fixed a bug where channel is being removed from my channel list when other member leaves the channel
- Fixed a bug where channel avatar image is not updated when a member leaves, or joins, or profileUrl changes

### After fix
Note the video displays first fix only.

https://github.com/sendbird/sendbird-uikit-react/assets/16806397/a729c835-569d-4ad5-95f8-193bf44b2fb1

Co-authored-by: @aldenquimby

[CLNP-2121]: https://sendbird.atlassian.net/browse/CLNP-2121?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/modules/ChannelList/dux/reducers.ts [hunk 1]
```diff
@@ -108,22 +109,61 @@ export default function channelListReducer(
...
+         *     2-1-1. If channel no longer matches the query
+         *       - Remove the channel from channel list
+         *       - Replace currentChannel with the next ordered channel
+         *     2-1-2. If channel matches the query:
+         *       - Upsert channel list with the channel
+         *       - Replace currentChannel IFF url is same
+         *   2-2. If query is not given,
+         *     - Upsert channel list with the channel
+         *     - Replace currentChannel IFF url is same
+         */
```
Comment (HoonBaek) [inline src/modules/ChannelList/dux/reducers.ts:132]: `1` and `2-1-1`, `2-1-2` and `2-2` have same action, can we reduce the lines?

Comment (liamcho) [reply src/modules/ChannelList/dux/reducers.ts:130]: yes it is done!

File: src/modules/ChannelList/dux/reducers.ts [hunk 2]
```diff
@@ -108,22 +109,57 @@ export default function channelListReducer(
...
             nextChannels = getChannelsWithUpsertedChannel(allChannels, channel, state.channelListQuery?.order);
+            if (currentChannel?.url === channel.url) {
+              nextChannel = channel;
+            }
+          }
+        } else {
+          nextChannels = getChannelsWithUpsertedChannel(allChannels, channel, state.channelListQuery?.order);
+          if (currentChannel?.url === channel.url) {
+            nextChannel = channel;
           }
```
Comment (HoonBaek) [inline src/modules/ChannelList/dux/reducers.ts:162]: How about reducing lines like this?
```suggestion
if (isMe || (channelListFilter && !filterChannelListParams(channelListQuery, channel, currentUserId))) {
const channelAt = allChannels.findIndex((ch: GroupChannel) => ch.url === channel.url);
          if (channelAt > -1) {
            nextChannels.splice(channelAt, 1);
            nextChannel = getNextChannel({
              channel,
              currentChannel,
              allChannels,
              disableAutoSelect,
            });
          }
} else {
nextChannels = getChannelsWithUpsertedChannel(allChannels, channel, state.channelListQuery?.order);
          if (currentChannel?.url === channel.url) {
            nextChannel = channel;
          }
}
```

**Review by HoonBaek — APPROVED** [review]
LGTM, however, pls fix the lint before merge :D
I removed the spaces in the comment;;

**Pull request merged**